src/project-manager.ts

```diff
@@ -321,6 +321,7 @@ export class ProjectManager implements Disposable {
 			span.addTags({ uri, maxDepth });
 			ignore.add(uri);
 			return this.ensureModuleStructure(span)
+				.concat(Observable.defer(() => this.ensureConfigDependencies()))
```

Comment:
Does this have a performance impact? I think it probably has, as you are iterating in `ensureConfigDependencies()` over every single file in the workspace, and `ensureReferencedFiles()` is called recursively for every referenced file, so you are repeating that iteration every time. I think `ensureConfigDependencies()` needs to memoize its return value in the same way `ensureAllFiles()` does, or you could just add that logic to `ensureModuleStructure()` (which would already is memoized and also has memoize cache invalidation logic).